### DISCUSSION TITLE
Get path for (file/temp)?

### DISCUSSION THREAD

**Discussion by neezer** [janet-lang/janet#1482] [Q&A] [answered]
If I make a temporary file with `(file/temp)`, how do I get the path to that file?

**Comment by sogaiu** [comment]
I think it's considered a feature that one cannot get a path:

```
    (file/temp)

    Open an anonymous temporary file that is removed on close. Raises 
    an error on failure.
```

**Reply by neezer** [reply]
Hmm. I was hoping to use this to create a temporary file, write to it, reference the path when calling an external command using [`janet-sh`](https://github.com/andrewchambers/janet-sh), then have that file automatically cleaned up when I closed it. Guess I can't?

**Reply by sogaiu** [reply, marked as answer]
I've also wanted automatic clean up but haven't found a nice way to accomplish it.

IIUC, the [current implementation uses `tmpfile`](https://github.com/janet-lang/janet/blob/631622aa485e7d947501623247d913ea9bd14324/src/core/io.c#L132):

```c
    FILE *tmp = tmpfile();
```

According to `tmpfile(3)`:

>       The tmpfile() function opens a unique temporary file in binary
>       read/write (w+b) mode.  The file will be automatically deleted when it
>       is closed or the program terminates.

It looks to me like it's the underlying standard C library that's being relied on to provide the clean up part as the man page has this bit near the top:

> LIBRARY
>       Standard C library (libc, -lc)
